Content.Server/Explosion/EntitySystems/ExplosionSystem.Processing.cs

```diff
@@ -0,0 +1,606 @@
+        IMapGrid grid,
+        Vector2i tile,
+        float throwForce,
+        DamageSpecifier damage,
+        MapCoordinates epicenter,
+        HashSet<EntityUid> processed,
+        string id)
+    {
+        var gridBox = new Box2(tile * grid.TileSize, (tile + 1) * grid.TileSize);
+
+        // get the entities on a tile. Note that we cannot process them directly, or we get
+        // enumerator-changed-while-enumerating errors.
+        List<(EntityUid, TransformComponent?) > list = new();
+        var xformQuery = EntityManager.GetEntityQuery<TransformComponent>();
+
+        EntityUidQueryCallback callback = uid =>
+        {
+            if (processed.Contains(uid))
+                return;
+
+            var xform = xformQuery.GetComponent(uid);
+
+            if (xform.ParentUid != grid.GridEntityId)
+            {
+                // Not parented to grid. Likely in a container.
+                if (_containerSystem.IsEntityInContainer(uid, xform))
+                    return;
+            }
+
+            list.Add((uid, xform));
+        };
+
+        _entityLookup.FastEntitiesIntersecting(lookup, ref gridBox, callback);
+
+        // process those entities
+        foreach (var (entity, xform) in list)
+        {
+            processed.Add(entity);
+            ProcessEntity(entity, epicenter, processed, damage, throwForce, id, false, xform);
+        }
+
+        // process anchored entities
+        var tileBlocked = false;
+        foreach (var entity in grid.GetAnchoredEntities(tile).ToList())
+        {
+            processed.Add(entity);
+            ProcessEntity(entity, epicenter, processed, damage, throwForce, id, true);
+            tileBlocked |= IsBlockingTurf(entity);
+        }
+
+        // Next, we get the intersecting entities AGAIN, but purely for throwing. This way, glass shards spawned from
+        // windows will be flung outwards, and not stay where they spawned. This is however somewhat unnecessary, and a
+        // prime candidate for computational cost-cutting. Alternatively, it would be nice if there was just some sort
+        // of spawned-on-destruction event that could be used to automatically assemble a list of new entities that need
+        // to be thrown.
+        //
+        // All things considered, until entity spawning & destruction is sped up, this isn't all that time consuming.
+        // And throwing is disabled for nukes anyways.
+        if (throwForce <= 0)
+            return !tileBlocked;
+
+        list.Clear();
+        _entityLookup.FastEntitiesIntersecting(lookup, ref gridBox, callback);
+
+        foreach (var (entity, xform) in list)
+        {
+            // Here we only throw, no dealing damage. Containers n such might drop their entities after being destroyed, but
+            // they handle their own damage pass-through.
+            ProcessEntity(entity, epicenter, processed, null, throwForce, id, false, xform);
+        }
+
+        return !tileBlocked;
```

Comment:
yeah the double-lookup to avoid weirdly not throwing some entities is a pretty brute force approach. but so far its still minor compared to all the destruction deletion+spawning.